save-core/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt

```diff
@@ -16,4 +18,12 @@ interface Plugin {
      * @return a sequence of [TestResult]s for each group of test resources
      */
     fun execute(saveProperties: SaveProperties, testConfig: TestConfig): Sequence<TestResult>
+
+    /**
+     * Discover groups of resource files which will be used to run tests.
+     *
+     * @param root root [Path], from where discovering should be started
+     * @return a sequence of files, grouped by test
+     */
+    fun discoverTestFiles(root: Path): Sequence<List<Path>>
```

Comment:
It seems, that return value actually is `Sequence<Pair<Path>>`, right?

May be we can explicitly specify this?

Reply:
No, `Pair` and `List` are not related, so they couldn't be cast to each other. We have a list with single element for `warn-plugin` and a list of two elements for `fix-plugin`, so to keep the API more general I've put a list here

Reply:
Ok, just saw something like `.map { it.first() to it.last() }`, so I thought, that we have the `Pair`. Also previously we had `discoverFilePairs`